common/util/user_interaction.cc

```diff
@@ -72,5 +82,21 @@ std::string inverse(absl::string_view s) {
   if (!IsInteractiveTerminalSession()) return std::string(s);
   return absl::StrCat(kInverseEscape, s, kNormalEscape);
 }
+
+absl::string_view StartColor(const std::ostream& stream, Color c) {
+  if (!IsInteractiveTerminalSession() || stream.rdbuf() != std::cout.rdbuf()) {
+    return "";
+  }
+  return kColorsStart[c];
+}
+
+absl::string_view EndColor(const std::ostream& stream, Color c) {
+  if (!IsInteractiveTerminalSession() || c == Color::kNone ||
+      stream.rdbuf() != std::cout.rdbuf()) {
```

Comment:
I like this idea of testing if the stream we're talking about is also indeed the stdout connection. I think this should be part of `IsInteractiveTerminalSession()`.

I've opened
https://github.com/chipsalliance/verible/pull/1870
to prepare that. After that is submitted, you'd need to rebase your change to that.

Reply:
Nice catch! :+1: